wcfsetup/install/files/lib/data/user/ignore/UserIgnoreAction.class.php

```diff
@@ -121,6 +152,126 @@ public function unignore()
         return ['isIgnoredUser' => 0];
     }
 
+    public function validateGetDialog()
+    {
+        $this->readInteger('userID');
+
+        $userProfile = UserProfileRuntimeCache::getInstance()->getObject($this->parameters['userID']);
+        if ($userProfile === null || $userProfile->userID == WCF::getUser()->userID) {
+            throw new IllegalLinkException();
+        }
+
+        $ignore = UserIgnore::getIgnore($this->parameters['userID']);
+
+        // Check if the user is not yet ignored and cannot be ignored.
+        if (!$ignore && $userProfile->getPermission('user.profile.cannotBeIgnored')) {
+            throw new PermissionDeniedException();
+        }
+    }
+
+    public function getDialog()
+    {
+        $form = $this->getForm();
+
+        return [
+            'dialog' => $form->getHtml(),
+            'formId' => $form->getId(),
+        ];
+    }
+
+    public function validateSubmitDialog()
+    {
+        $this->validateGetDialog();
+
+        $this->readString('formId');
+
+        $this->getForm()->requestData($this->parameters['data'] ?? []);
+        $this->getForm()->readValues();
+    }
+
+    public function submitDialog()
+    {
+        $this->getForm()->validate();
```

Comment:
Move this line to the end of `validateSubmitDialog()`.

Reply:
Yes, because the return value inside the condition has to be returned in `submitDialog()`.